### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by doomspork**
@hassox this fixes the warnings on master so that the Travis build passes.

### CODE REVIEW

File: lib/guardian/permissions.ex [hunk 1]
```diff
@@ -170,9 +170,6 @@ defmodule Guardian.Permissions do
     Map.get(c, type, Map.get(c, to_string(type), 0))
   end
 
-  @doc false
```
Comment (doomspork) [inline lib/guardian/permissions.ex]: Warning: 
```shell
lib/guardian/permissions.ex:174: warning: this clause cannot match because a 
previous clause at line 168 always matches
```

File: lib/guardian/plug/verify_header.ex [hunk 2]
```diff
@@ -35,7 +35,6 @@ defmodule Guardian.Plug.VerifyHeader do
       # will take the first auth header
       # Authorization: <jwt>
   """
-  import Guardian.Keys
```
Comment (doomspork) [inline lib/guardian/plug/verify_header.ex]: Warning: 
```shell
lib/guardian/plug/verify_header.ex:38: warning: unused import Guardian.Keys
```

File: lib/guardian/claims.ex [hunk 3]
```diff
@@ -110,7 +110,7 @@ defmodule Guardian.Claims do
         Map.put(claims, "exp", iat + years * 365 * 24 * 60 * 60)
       {iat, {years, :year}} ->
         Map.put(claims, "exp", iat + years * 365 * 24 * 60 * 60)
-      {iat, {_, units}} -> raise "Unknown Units: #{units}"
+      {_iat, {_, units}} -> raise "Unknown Units: #{units}"
```
Comment (doomspork) [inline lib/guardian/claims.ex]: Warning: 
```shell
lib/guardian/claims.ex:113: warning: variable iat is unused
```

File: lib/guardian/channel.ex [hunk 4]
```diff
@@ -37,14 +37,13 @@ defmodule Guardian.Channel do
   directly and authenticating the connection rather than the channel.
   """
   defmacro __using__(opts) do
-    opts = Enum.into(opts, %{})
-    key = Map.get(opts, :key, :default)
+    key = Keyword.get(opts, :key, :default)
```
Comment (doomspork) [inline lib/guardian/channel.ex]: Warning:

```shell
lib/guardian/channel.ex:41: warning: variable key is unused
```

@hassox this was probably a bug, key wasn't being unquoted.  Can you review this change?

**Comment by aaronrenner** [comment]
Looks great to me. :+1: This will be super helpful so my tests will pass.

File: lib/guardian/channel.ex (hunk 4 above)
Comment (hassox) [inline lib/guardian/channel.ex]: Why the change to Keyword from map? I think this will impact other areas of the code.

Comment (hassox) [inline lib/guardian/channel.ex]: Nm

Comment (doomspork) [inline lib/guardian/channel.ex]: @hassox I made this change because, at first glance, it didn't look like `opts` was being used other than retrieving `key` and converting to a map seemed like overkill.
